render.go

```diff
@@ -77,37 +80,41 @@ func (r *Renderer) DrawShapedRunAt(run shaping.Output, img draw.Image, startX, s
 		r.PixScale = 1
 	}
 	scale := r.FontSize * r.PixScale / float32(run.Face.Upem())
+	r.fillerScale = scale
 
 	b := img.Bounds()
 	scanner := rasterx.NewScannerGV(b.Dx(), b.Dy(), img, b)
 	f := rasterx.NewFiller(b.Dx(), b.Dy(), scanner)
+	r.filler = f
 	f.SetColor(r.Color)
 	x := float32(startX)
 	y := float32(startY)
 	for _, g := range run.Glyphs {
 		x -= fixed266ToFloat(g.XOffset) * r.PixScale
```

Comment:
Could you explain why is it a substraction ? I would rather have added the offset.

The shaper assume that rendering is done by, given a dot at x,y :

- draw the glyph at x + XOffset,y + YOffset 
- move the dot to x + XAdvance, y + YAdvance
- go to the next glyph and repeat

Reply:
> * draw the glyph at x + XOffset,y + YOffset

Isn't the Y advance negative in this context? The height certainly is.

Reply:
> Isn't the Y advance negative in this context? The height certainly is.

For vertical text, yes, the YAdvance would generally by negative (following the top to bottom axis convention).
It is actually ignored for horizontal text.
(YOffset and XOffset have unknown sign in general.)

Reply:
That's good to know. I did mean Y offset however - does the same thing apply?

Reply:
> That's good to know. I did mean Y offset however - does the same thing apply?

Yes, the same convention applies. (but you cant predict if `YOffset` should be positive or negative since a glyph may be moved up or down.)